### PR TITLE
Create a cache for snippets

### PR DESCRIPTION
**Pull request by ccordoba12**
- This builds on top of the excellent work done by @krassowski on PR #26 to build a cache for snippets.
- As with labels, this increases completions speed dramatically.

### CODE REVIEW

**Comment by krassowski** [comment]
Looks good to me. I won't have time to properly test it until next week though.

**Comment by ccordoba12** [comment]
No prob, I can wait for your review.

**Comment by ccordoba12** [comment]
Hey @krassowski, any news on this?

**Comment by krassowski** [comment]
Sorry, not yet! Feel free to merge - I will take time to look at it ultimately and create a follow up PR if find anything in need of improvement.

**Review by krassowski — APPROVED** [review]

File: CONFIGURATION.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ This server can be configured using `workspace/didChangeConfiguration` method. E
 | `pylsp.plugins.jedi_completion.include_class_objects` | `boolean` | Adds class objects as a separate completion item. | `true` |
 | `pylsp.plugins.jedi_completion.fuzzy` | `boolean` | Enable fuzzy when requesting autocomplete. | `false` |
 | `pylsp.plugins.jedi_completion.eager` | `boolean` | Resolve documentation and detail eagerly. | `false` |
-| `pylsp.plugins.jedi_completion.resolve_at_most_labels` | `number`  | How many labels (at most) should be resolved? | `25` |
-| `pylsp.plugins.jedi_completion.cache_labels_for` | `array`  of  `string` items | Modules for which the labels should be cached. | `["pandas", "numpy", "tensorflow", "matplotlib"]` |
+| `pylsp.plugins.jedi_completion.resolve_at_most` | `number`  | How many labels and snippets (at most) should be resolved? | `25` |
```
Comment (andfoy) [inline CONFIGURATION.md:25]: Isn't this a BC change?

Comment (ccordoba12) [reply CONFIGURATION.md:25]: What do you mean?

Comment (andfoy) [reply CONFIGURATION.md:25]: We need to document this backwards compatibility difference somewhere to let clients that these flags have changed names.

Comment (ccordoba12) [reply CONFIGURATION.md:25]: Ah, ok, I totally agree with that. Will do it in our release notes when 1.3.0 is out.

**Comment by stonebig** [comment]
any hope to have this soon ?

**Comment by ccordoba12** [comment]
It will come along with Spyder 5.2.0, to be released in November.

**Pull request merged**